service/telemetry.go

```diff
@@ -109,8 +110,29 @@ func (tel *colTelemetry) initOnce(col *Collector) error {
 
 	logger.Info("Setting up own telemetry...")
 
-	instanceUUID, _ := uuid.NewRandom()
-	instanceID := instanceUUID.String()
+	// Construct telemetry attributes from resource attributes.
+	telAttrs := map[string]string{}
+	for k, v := range resource {
+		// nil value indicates that the attribute should not be included in the telemetry.
+		if v != nil {
+			telAttrs[k] = *v
+		}
+	}
+
+	if _, ok := resource[semconv.AttributeServiceInstanceID]; !ok {
+		// AttributeServiceInstanceID is not specified in the config. Auto-generate one.
+		instanceUUID, _ := uuid.NewRandom()
+		instanceID := instanceUUID.String()
+		telAttrs[semconv.AttributeServiceInstanceID] = instanceID
+	}
+
+	if AddCollectorVersionTag {
+		if _, ok := resource[semconv.AttributeServiceVersion]; !ok {
+			// AttributeServiceVersion is not specified in the config. Use the actual
+			// build version.
+			telAttrs[semconv.AttributeServiceVersion] = version.Version
```

Comment:
On non-core distros, this will show `latest`, right? We probably want to explicitly pass a version on the Collector settings to avoid this

Reply:
I haven't verified, but I think you are right. This PR didn't change the behavior.

Reply:
You are right, I missed that this was not introduced by this PR. I will make a note to reproduce and open an issue then. We probably want to get rid of `version.Version` entirely and leave passing the version to the builder instead.

Reply:
There is another semi-related issue: for Agent Management the Supervisor [needs a way](https://docs.google.com/document/d/1KtH5atZQUs9Achbce6LiOaJxLbksNJenvgvyKLsJrkc/edit#heading=h.ioikt02qpy5f) to fetch the Collector's version.

If we are going to change how versions are embedded in the Collector I want to use the opportunity to see if we can make it fetchable by the Supervisor. A few options:
- Have a manifest file with version number recorded in it and include the manifest in Collector's installation packages.
- Try to fetch version number from executable directly (as a symbol?). Not very desirable since it requires platform specific code.
- Read from CLI by using `--version` flag (not very reliable since the output format is not precisely defined).
- Something else?

Reply:
I submitted a separate issue for this https://github.com/open-telemetry/opentelemetry-collector/issues/5438